Certain malformed NTLMSSP packets could crash the NTLM helpers provided by Squid. 
This patch fixes this security issue, that is described at Squid Bugzilla #1045;
http://www.squid-cache.org/bugs/show_bug.cgi?id=1045

--- squid-2.5.STABLE5/helpers/ntlm_auth/fakeauth/fakeauth_auth.c			2001-08-29 16:57:38.000000000 +0200
+++ squid-2.5.STABLE5/helpers/ntlm_auth/fakeauth/fakeauth_auth.c.ntlm_fetch_string	2004-09-03 19:20:52.000000000 +0200
@@ -150,7 +150,7 @@
     o = WSWAP(str->offset);
 
     /* Sanity checks. XXX values arbitrarialy chosen */
-    if (l <= 0 || l >= 32 || o >= 256) {
+    if (l <= 0 || o <= 0 || l >= 32 || o >= 256) {
 	fprintf(stderr, "ntlmGetString: insane: l:%d o:%d\n", l, o);
 	return (NULL);
     }
--- squid-2.5.STABLE5/lib/ntlmauth.c							2002-07-22 04:06:31.000000000 +0200
+++ squid-2.5.STABLE5/lib/ntlmauth.c.ntlm_fetch_string					2004-09-03 19:21:44.000000000 +0200
@@ -84,7 +84,7 @@
     o = WSWAP(str->offset);
     /* debug("fetch_string(plength=%d,l=%d,o=%d)\n",length,l,o); */
 
-    if (l < 0 || l > MAX_FIELD_LENGTH || o + l > length || o == 0) {
+    if (l < 0 || o <= 0 || l > MAX_FIELD_LENGTH || o + l > length) {
 	/* debug("ntlmssp: insane data (l: %d, o: %d)\n", l,o); */
 	return rv;
     }
